File: files/CSEC 02032025.R
library(tidyverse)
library(readxl)
library(lubridate)
library(hms)

path = "files/Ex-Challenge 09 2025.xlsx"
input = read_excel(path, range = "B3:B7")
test  = read_excel(path, range = "D3:F7") %>%
  mutate(Time = as_hms(Time) %>% as.POSIXct())

result = input %>%
  mutate(Dates = gsub("\\.", "", Dates)) %>%
  mutate(Dates = parse_date_time(Dates, "b. d, Y, I:M p")) %>%
  mutate(Date = as.Date(Dates) %>% as.POSIXct(),
         Day = wday(Dates, label = TRUE, abbr = FALSE, locale = "en") %>% as.character(),
         Time = as_hms(Dates) %>% as.POSIXct()) %>%
  select(-Dates)

all.equal(result, test, check.attributes = FALSE)
# [1] TRUE

File: files/2025-03-02/CSEC 02032025.py
import pandas as pd

path = "files/Ex-Challenge 09 2025.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=2, nrows=5)
test = pd.read_excel(path, usecols="D:F", skiprows=2, nrows=5)

input['Dates'] = input['Dates'].apply(lambda x: x.split()[0][:3] + ' ' + ' '.join(x.split()[1:]))
input['Dates'] = input['Dates'].str.replace('.', '', regex=False)
input['Dates'] = input['Dates'].str.replace('am', 'AM').str.replace('pm', 'PM')

input['Date'] = pd.to_datetime(input['Dates']).dt.date.astype('datetime64[ns]')
input['Day'] = pd.to_datetime(input['Dates']).dt.day_name()
input['Time'] = pd.to_datetime(input['Dates']).dt.time

input = input.drop(columns=['Dates'])

print(input.equals(test)) # True
